purchases/src/main/kotlin/com/revenuecat/purchases/deprecatedListenerConversions.kt

```diff
@@ -0,0 +1,20 @@
+package com.revenuecat.purchases
+
+/**
+ * Get latest available purchaser info.
+ * @param onSuccess Called when purchaser info is available and not stale. Called immediately if
+ * purchaser info is cached.
+ * @param onError Will be called after the call has completed with an error.
+ */
+@Suppress("unused")
+@Deprecated(
+    "Renamed to getCustomerInfoWith",
+    level = DeprecationLevel.ERROR,
+    replaceWith = ReplaceWith("Purchases.sharedInstance.getCustomerInfoWith(onError, onSuccess)")
+)
+fun Purchases.getPurchaserInfoWith(
+    onError: ErrorFunction = ON_ERROR_STUB,
+    onSuccess: (purchaserInfo: PurchaserInfo) -> Unit
+) {
+    // No-op
+}
```

Comment:
Since this is a Kotlin extension function it can be moved to its own file

Reply:
Something cool is that since these Kotlin conversion functions use lambdas, most of the people won't be referencing the type of the PurchaserInfo directly so it should work. Like this example should still work after the rename:

```
Purchases.sharedInstance.restorePurchasesWith { info ->
            
}
```

If someone were doing something like:

```
Purchases.sharedInstance.restorePurchasesWith { info: PurchaserInfo ->
            
}
```

Android Studio would fail the compilation and it would also indicate that `PurchaserInfo` has been renamed.

We can't add overloaded functions to add deprecations for these functions because for Kotlin, changing a lambda parameter doesn't change the signature and it's not possible to do:

```
fun Purchases.restorePurchasesWith(
    onError: ErrorFunction = ON_ERROR_STUB,
    onSuccess: (customerInfo: CustomerInfo) -> Unit
) {
    restorePurchases(receiveCustomerInfoListener(onSuccess, onError))
}

@Deprecated
fun Purchases.restorePurchasesWith(
    onError: ErrorFunction = ON_ERROR_STUB,
    onSuccess: (customerInfo: PurchaserInfo) -> Unit
) {
    restorePurchases(receiveCustomerInfoListener(onSuccess, onError))
}
```

Reply:
That's basically the exact same with Swift 🎉

Reply:
it's _possible_ these were called from Java, i guess that's pretty unlikely though and we aren't going to worry about it?

Reply:
Yes, I wouldn't worry about it. I doubt anyone is doing that since they are kinda awkward to call from Java